File: app/src/main/java/sgtmelon/scriptum/app/model/RankRepo.java
package sgtmelon.scriptum.app.model;

import java.util.List;

import sgtmelon.scriptum.app.model.item.RankItem;

/**
 * Репозиторий категории
 */
public final class RankRepo {

    private final List<String> listName;
    private List<RankItem> listRank;

    public RankRepo(List<RankItem> listRank, List<String> listName) {
        this.listRank = listRank;
        this.listName = listName;
    }

    public List<String> getListName() {
        return listName;
    }

    public List<RankItem> getListRank() {
        return listRank;
    }

    public void setListRank(List<RankItem> listRank) {
        this.listRank = listRank;
    }

    public int size() {
        return listRank.size();
    }

    public void add(int position, RankItem rankItem) {
        listRank.add(position, rankItem);
        listName.add(position, rankItem.getName().toUpperCase());
    }

    public void remove(int position) {
        listRank.remove(position);
        listName.remove(position);
    }

    public void move(int positionOld, int positionNew) {
        final RankItem rankItem = listRank.get(positionOld);

        remove(positionOld);
        add(positionNew, rankItem);
    }

}

File: app/src/main/java/sgtmelon/scriptum/app/model/RankRepo.kt
package sgtmelon.scriptum.app.model

import sgtmelon.scriptum.app.model.item.RankItem

/**
 * Репозиторий категории
 */
class RankRepo(val listRank: MutableList<RankItem>,
               val listName: MutableList<String>
) {

    fun setListRank(listRank: List<RankItem>) {
        this.listRank.clear()
        this.listRank.addAll(listRank)
    }

    fun size(): Int = listRank.size

    fun set(position: Int, name: String) {
        listRank[position].name = name
        listName[position] = name.toUpperCase()
    }

    fun add(position: Int, rankItem: RankItem) {
        listRank.add(position, rankItem)
        listName.add(position, rankItem.name.toUpperCase())
    }

    fun remove(position: Int) {
        listRank.removeAt(position)
        listName.removeAt(position)
    }

    fun move(positionFrom: Int, positionTo: Int) {
        val rankItem = listRank[positionFrom]

        remove(positionFrom)
        add(positionTo, rankItem)
    }

}